tools/getrealdeps.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+# getrealdeps for Chromebrew
+# Author: Satadru Pramanik (satmandu) satadru at gmail dot com
+# set -x
+pkg="${1}"
+
+RED='\e[1;91m';    # Use Light Red for errors.
+YELLOW='\e[1;33m'; # Use Yellow for informational messages.
+GREEN='\e[1;32m';  # Use Green for success messages.
+BLUE='\e[1;34m';   # Use Blue for intrafunction messages.
+GRAY='\e[0;37m';   # Use Gray for program output.
+MAGENTA='\e[1;35m';
+RESET='\e[0m'
+
+# Simplify colors and print errors to stderr (2).
+echo_error() { echo -e "${RED}${*}${RESET}" >&2; }
+echo_info() { echo -e "${YELLOW}${*}${RESET}" >&1; }
+echo_success() { echo -e "${GREEN}${*}${RESET}" >&1; }
+echo_intra() { echo -e "${BLUE}${*}${RESET}" >&1; }
+echo_out() { echo -e "${GRAY}${*}${RESET}" >&1; }
+echo_other() { echo -e "${MAGENTA}${*}${RESET}" >&2; }
+
+# Install grep if a functional local copy does not exist.
+if grep --version &> /dev/null; then
+  GREP=grep
+else
+  crew install grep
+  GREP=/usr/local/bin/grep
+fi
+
+: "${CREW_LIB_PREFIX:=$(crew const | $GREP CREW_LIB_PREFIX | awk -F = '{print $2}')}"
+
+# Which packages have a needed library in CREW_LIB_PREFIX
+# This is a subset of what crew whatprovides gives
+whatprovidesfxn() {
+  pkgdepslcl="${1}"
+  filelcl=$($GREP --exclude "${pkg}.filelist" "$pkgdepslcl" /usr/local/etc/crew/meta/*.filelist | $GREP "$CREW_LIB_PREFIX")
+  packagelcl=$(echo "$filelcl" | \
+  sed 's/.filelist.*//g' | sed 's:.*/::' | awk '!x[$0]++' | sed s/://g)
+  echo "$packagelcl"
+}
+
+# What files does a package provide
+crewfilesfxn() {
+ pkgname="${1}"
+ files=$(< /usr/local/etc/crew/meta/"${pkgname}".filelist)
+ echo "$files"
+}
+
+pkgfiles=$(crewfilesfxn "${pkg}")
+# Use readelf to determine library dependencies, as
+# this doesn't almost run a program like using ldd would.
```

Comment:
`ldd` (or `LD_TRACE_LOADED_OBJECTS`) would be faster since it doesn't parse the entire binary:
```shell
$ time readelf -d /usr/local/bin/ruby

Dynamic section at offset 0x3028 contains 38 entries:
  Tag        Type                         Name/Value
 0x0000000000000001 (NEEDED)             Shared library: [libruby.so.3.1]
 0x0000000000000001 (NEEDED)             Shared library: [libz.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libpthread.so.0]
 0x0000000000000001 (NEEDED)             Shared library: [librt.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libgmp.so.10]
 0x0000000000000001 (NEEDED)             Shared library: [libdl.so.2]
 0x0000000000000001 (NEEDED)             Shared library: [libcrypt.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libm.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [ld-linux-x86-64.so.2]
...snip...
real    0m0.029s
user    0m0.025s
sys     0m0.004s
```
```shell
$ time LD_TRACE_LOADED_OBJECTS=1 /usr/local/bin/ruby
        linux-vdso.so.1 (0x00007ffeac8b9000)
        libruby.so.3.1 => /usr/local/lib64/libruby.so.3.1 (0x00007dc854df4000)
        libz.so.1 => /usr/local/lib64/libz.so.1 (0x00007dc854dd7000)
        libpthread.so.0 => /usr/local/lib64/libpthread.so.0 (0x00007dc854db6000)
        librt.so.1 => /usr/local/lib64/librt.so.1 (0x00007dc854dab000)
        libgmp.so.10 => /usr/local/lib64/libgmp.so.10 (0x00007dc854d08000)
        libdl.so.2 => /usr/local/lib64/libdl.so.2 (0x00007dc854d01000)
        libcrypt.so.1 => /usr/local/lib64/libcrypt.so.1 (0x00007dc854cc6000)
        libm.so.6 => /usr/local/lib64/libm.so.6 (0x00007dc854bc2000)
        libc.so.6 => /usr/local/lib64/libc.so.6 (0x00007dc854a03000)
        /lib64/ld-linux-x86-64.so.2 (0x00007dc8551c2000)
        libgcc_s.so.1 => /usr/local/lib64/libgcc_s.so.1 (0x00007dc8549e3000)

real    0m0.005s
user    0m0.001s
sys     0m0.005s
```

Reply:
Yes, but readelf is safer. As I understand it ldd almost runs the binary...

https://news.ycombinator.com/item?id=8508628